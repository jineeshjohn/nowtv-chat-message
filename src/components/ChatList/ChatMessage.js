import React, { Component } from "react";
import { PropTypes } from "prop-types";
import "./style.css";
import defaultAvatar from "./../../resources/avatar.png";

class ChatMessage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showEmail: false
    };
    this.mouseEnter = this.mouseEnter.bind(this);
    this.mouseLeave = this.mouseLeave.bind(this);
  }
  mouseEnter() {
    this.setState({
      showEmail: true
    });
  }
  mouseLeave() {
    this.setState({
      showEmail: false
    });
  }
  render() {
    const { message, fullName, timestamp, avatar, email } = this.props;
    const { showEmail } = this.state;
    return (
      <div
        className="container"
        onMouseEnter={this.mouseEnter}
        onMouseLeave={this.mouseLeave}
      >
        <div>
          <img src={avatar || defaultAvatar} alt="Avatar" />
          <div>
            <div className="email">{showEmail && email}</div>
            <div className="fullname">{fullName}</div>
            <div className="message">{message}</div>
          </div>
        </div>
        <span className="time-right">{timestamp}</span>
      </div>
    );
  }
}

ChatMessage.propTypes = {
  messageId: PropTypes.string,
  userId: PropTypes.string,
  fullName: PropTypes.string,
  timestamp: PropTypes.string,
  email: PropTypes.string,
  message: PropTypes.string,
  avatar: PropTypes.string
};

ChatMessage.defaultProps = {
  avatar: null
};

export default ChatMessage;
