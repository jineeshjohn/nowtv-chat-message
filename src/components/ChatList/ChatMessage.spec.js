import React from "react";
import { shallow } from "enzyme";
import ChatMessage from "./ChatMessage";

describe("<ChatMessage>", () => {
  let wrapper = null;
  const props = {
    messageId: "12356",
    userId: "613651251",
    fullName: "Robin Balmforth",
    timestamp: "2016-06-03T20:24:29Z",
    email: "robin@example.com",
    message: "Hello, World!",
    avatar: null
  };

  beforeEach(() => {
    wrapper = shallow(<ChatMessage {...props} />);
    jest.spyOn(ChatMessage.prototype, "mouseEnter");
    jest.spyOn(ChatMessage.prototype, "mouseLeave");
  });
  afterEach(() => {
    wrapper = null;
  });

  it("render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have one container", () => {
    expect(wrapper.find(".container").length).toEqual(1);
  });
  it("should have one email", () => {
    expect(wrapper.find(".email").length).toEqual(1);
  });
  it("should have one fullname", () => {
    expect(wrapper.find(".fullname").length).toEqual(1);
  });
  it("should have one avatar", () => {
    expect(wrapper.find("img").length).toEqual(1);
  });
  it("should have one message", () => {
    expect(wrapper.find(".message").length).toEqual(1);
  });
  it("should simulate mouseEnter and update the state", () => {
    wrapper.simulate("mouseEnter");
    expect(ChatMessage.prototype.mouseEnter).toHaveBeenCalled();
    expect(wrapper.state().showEmail).toBe(true);
  });
  it("should simulate mouseLeave and update the state", () => {
    wrapper.simulate("mouseLeave");
    expect(ChatMessage.prototype.mouseLeave).toHaveBeenCalled();
    expect(wrapper.state().showEmail).toBe(false);
  });
});
