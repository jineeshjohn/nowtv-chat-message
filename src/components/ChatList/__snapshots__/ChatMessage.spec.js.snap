// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMessage> render without crashing 1`] = `
<div
  className="container"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div>
    <img
      alt="Avatar"
      src=""
    />
    <div>
      <div
        className="email"
      />
      <div
        className="fullname"
      >
        Robin Balmforth
      </div>
      <div
        className="message"
      >
        Hello, World!
      </div>
    </div>
  </div>
  <span
    className="time-right"
  >
    2016-06-03T20:24:29Z
  </span>
</div>
`;
